### PR TITLE
initial clean of aero_model.F90

### PR DESCRIPTION
**Pull request by tangshuaiqi**
Initial clean up of file aero_model.F90:

1. remove dead code and unused variables
2. remove options related to MMF
3. remove logical conditions that are always .true.
4. clean up module variables

### CODE REVIEW

**Review by mingxuanwupnnl — APPROVED** [review]

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 1]
```diff
@@ -259,40 +228,17 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
     !    (1) All of the "if ( convproc_do_aer .or. convproc_do_gas ) then" statements 
     !        in aero_model.F90 have been changed to "if ( convproc_do_aer ) then"
     !    (2) convproc_do_aer=.false. and convproc_do_gas=.true. is no longer allowed.
-    if ( ( .not. convproc_do_aer ) .and. convproc_do_gas ) then
-       errmes = 'aero_model_init - ' // &
-          'convproc_do_aer MUST BE .true. when convproc_do_gas is .true.' 
+    ! for C++ porting: All convproc_do_aer=.false. and resus_fix=.false. conditions
+    ! are removed as not been used/tested. These conditions are then not allowed.
+    ! convproc_do_gas is not not used
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: should there be two `not`s here?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 2]
```diff
@@ -644,276 +578,7 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
...
-    index_chm_mass(3,1) = get_spc_ndx('so4_a3')
-    !
-    ! POM mode
-    !
-    index_tot_mass(4,1) = get_spc_ndx('pom_a4')
-    index_tot_mass(4,2) = get_spc_ndx('bc_a4')
-    index_tot_mass(4,3) = get_spc_ndx('mom_a4')
-    index_chm_mass(4,1) = get_spc_ndx('bc_a1' )
-    !
-#elif ( defined MODAL_AERO_4MODE )
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/aero_model.F90:742]: Do we want to get rid of this 4-mode code? Are we not using `index_tot_mass` and `index_chm_mass` anymore?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 3]
```diff
@@ -1131,42 +787,13 @@ subroutine aero_model_drydep  ( state, pbuf, obklen, ustar, cam_in, dt, cam_out,
...
-             qaerwat(1:ncol,:,mm) = qaerwat(1:ncol,:,mm) + dqdt_tmp(1:ncol,:) * dt
-
           else  ! lphase == 2
              ! use pvprogseasalts instead (means making the top level 0)
              pvmzaer(:ncol,1)=0._r8
              pvmzaer(:ncol,2:pverp) = vlc_dry(:ncol,:,jvlc)
              fldcw => qqcw_get_field(pbuf, mm,lchnk)
 
-             if(.true.) then ! use phil's method
+!             if(.true.) then ! use phil's method
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/aero_model.F90:796]: if you leave this in as a comment, why not leave the "charlie's method" code below as comments, too?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 4]
```diff
@@ -2586,23 +1911,11 @@ subroutine modal_aero_bcscavcoef_init
           xxfitvol(1,nnfit) = 1._r8
           yyfitvol(nnfit) = log( scavratevol )
 
-5900      continue
-
           !
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: We can probably delete these comments.

**Review by jeff-cohere — APPROVED** [review]
Looks good. Just a few questions/comments.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: We can change this line as: `if (.not. (convproc_do_aer .and. resus_fix)) then`
Do you mean this?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 2 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:742]: we are not using those indexes so I remove all these lines.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (jeff-cohere) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: I was referring to `! convproc_do_gas is not not used`, thinking maybe it means `! convproc_do_gas is not used`

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 3 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:796]: I leave this comment just to locate the code that can be separated into a subroutine as it is used a few times. Will clean it when I refactor this subroutine.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 4 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: will do it.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: aha, thank you for catching it. I will remove one

**Comment by tangshuaiqi** [comment]
All tests passed. merge into maint

  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**